### ISSUE TITLE
XMLHttpRequest.open doesn't get generated properly

### ISSUE THREAD

**Issue by jackalcooper** [OvermindDL1/bucklescript-tea#19]
bucklescript-tea/src/web_xmlhttprequest.ml:
```ocaml
let open_ method' url ?(async=true) ?(user="") ?(password="") x =
      x##open_ method' url async user password
```
```javascript
function open_(method$prime, url, $staropt$star, $staropt$star$1, $staropt$star$2, x) {
  var async = $staropt$star ? $staropt$star[0] : /* true */1;
  var user = $staropt$star$1 ? $staropt$star$1[0] : "";
  var password = $staropt$star$2 ? $staropt$star$2[0] : "";
  return x.open_(method$prime, url, async, user, password);
}
```
Which should be:
```
xhrReq.open(method, url);
xhrReq.open(method, url, async);
xhrReq.open(method, url, async, user);
xhrReq.open(method, url, async, user, password);
```

**Comment by OvermindDL1** [comment]
I've not actually made the TEA interface for requests as ELM's version was in a big flux until just recently and I wanted to match it to make conversion easier.  I'll be looking at making the Elm-style interface when I get time, unless you want to PR it first?  ^.^

But yeah, due to it being untested that above code is completely and entirely untested.  ^.^;

**Comment by jackalcooper** [comment]
Actually I've PR already.https://github.com/OvermindDL1/bucklescript-tea/pull/20

**Comment by jackalcooper** [comment]
Here is another runtime exception:
```
Uncaught TypeError: x.send_string is not a function
    at Object.send (web_xmlhttprequest.js:636)
    at tea_http.js:728
    at Object._1 (curry.js:79)
    at Object.run (tea_cmd.js:45)
    at handler (tea_app.js:138)
    at Object._2 (curry.js:140)
    at handler (tea_app.js:36)
    at Object._1 (curry.js:79)
    at HTMLButtonElement.<anonymous> (vdom.js:261)
```
```javascript
switch (body.tag | 0) {
      case 0 : 
          return x.send_string(body[0]);
      case 1 : 
          return x.send_formdata(body[0]);
      case 2 : 
          var form = List.fold_left(function (f, param) {
                Web_formdata.append(param[0], param[1], f);
                return f;
              }, new FormData(), body[0]);
          return x.send_formdata(form);
      case 3 : 
```

**Comment by bobzhang** [comment]
this is the same issue we saw before, maybe you need do a upgrade to `send__string` etc

**Comment by jackalcooper** [comment]
Yes

**Comment by OvermindDL1** [comment]
Definitely `send__string` would be best here, thanks for the PR, merged and published to 0.3.0.  :-)

Keep them coming!  :-)
